module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -604,13 +611,54 @@ public function getHolding($bibId, array $patron = null, array $options = [])
                     $item->itemLevelCallNumberPrefix ?? '',
                     $item->itemLevelCallNumber ?? ''
                 );
+
+                $dueDateValue = '';
+                if ($item->status->name == 'Checked out'
+                    && $showDueDate
+                    && $dueDateItemCount < $maxNumDueDateItems
+                ) {
+                    // extract itemId & construct query
+                    $query = [
```

Comment:
It might improve readability to extract the logic here to a `protected getDueDate($item, $showTime)` method. It will not only make this function shorter and more readable, but it will also make it easier for people to customize the due date retrieval in local custom code if they wish to.

Reply:
Very fair point and very true. I'll commit code soon, would appreciate your review and your testing in your live system, I did test those changes, of course, but you know how it goes - more eyes = more better (sorry for the grammar)